src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -357,25 +357,37 @@ StorageReplicatedMergeTree::StorageReplicatedMergeTree(
     /// It does not make sense for CREATE query
     if (attach)
     {
-        if (current_zookeeper && current_zookeeper->exists(replica_path + "/host"))
+        try
```

Comment:
When I added AttachThread I left this part as a possible failure spot because of `force_restore_data`, I don't think we check it again after this point.
But it will be ignored also if we can't create a ZK connection while attaching so this should be consistent with that behavior.

In general, now we basically ignore the flag if the ZK connection is not stable, are we okay with that?

Reply:
I think it's totally fine to ignore this flag in case of connection issues. In the worst case we will fail to load data parts due to sanity checks, the server will stop and will try on the next start. Before this change it would stop too, but due to a connection issue, so the only thing that changed is the error message. 

As for `checkHasReplicaMetadataInZooKeeper`, it is checked in the attach thread as well.